pos_loyalty_exclude/models/loyalty_rule.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2024 Camptocamp (https://www.camptocamp.com).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.osv import expression
+
+
+class LoyaltyRule(models.Model):
+    _inherit = "loyalty.rule"
+
+    @api.depends("product_ids", "product_category_id", "product_tag_id")
+    def _compute_valid_product_ids(self):
+        super()._compute_valid_product_ids()
+        # exclude product when no config in loyalty rule
```

Comment:
I think it would be worth adding a condition here to only proceed if we actually have such products ``self.env["product.product"].search_count[("loyalty_exclude", "=", True)]``

otherwise when we install the module, we introduce a small performance regression due to `any_product = False`

Reply:
Indeed, ftr, I tried to patch the way it calculates the loyalty but they are quite tightly-coupled